File: filesystem/src/main/java/com/nytimes/android/external/fs/RecordPersister.java
package com.nytimes.android.external.fs;

import com.nytimes.android.external.fs.filesystem.FileSystem;
import com.nytimes.android.external.store.base.RecordProvider;
import com.nytimes.android.external.store.base.RecordState;
import com.nytimes.android.external.store.base.impl.BarCode;

import java.util.concurrent.TimeUnit;

import javax.annotation.Nonnull;
import javax.inject.Inject;

public class RecordPersister extends SourcePersister implements RecordProvider<BarCode> {

    @Nonnull
    private final TimeUnit expirationUnit;
    private final long expirationDuration;

    @Inject
    public RecordPersister(FileSystem fileSystem,
                           long expirationDuration,
                           @Nonnull TimeUnit expirationUnit) {
        super(fileSystem);
        this.expirationDuration = expirationDuration;
        this.expirationUnit = expirationUnit;
    }

    public static RecordPersister create(FileSystem fileSystem,
                                         long expirationDuration,
                                         @Nonnull TimeUnit expirationUnit) {
        return new RecordPersister(fileSystem, expirationDuration, expirationUnit);
    }

    @Nonnull
    @Override
    public RecordState getRecordState(@Nonnull BarCode barCode) {
        return sourceFileReader.getRecordState(barCode, expirationUnit, expirationDuration);
    }
}

File: filesystem/src/main/java/com/nytimes/android/external/fs3/RecordPersister.kt
package com.nytimes.android.external.fs3

import com.nytimes.android.external.fs3.filesystem.FileSystem
import com.nytimes.android.external.store3.base.RecordProvider
import com.nytimes.android.external.store3.base.RecordState
import com.nytimes.android.external.store3.base.impl.BarCode

import java.util.concurrent.TimeUnit
import javax.inject.Inject

class RecordPersister @Inject
constructor(fileSystem: FileSystem,
            private val expirationDuration: Long,
            private val expirationUnit: TimeUnit) : SourcePersister(fileSystem), RecordProvider<BarCode> {

    override fun getRecordState(key: BarCode): RecordState {
        return sourceFileReader.getRecordState(key, expirationUnit, expirationDuration)
    }

    companion object {

        fun create(fileSystem: FileSystem,
                   expirationDuration: Long,
                   expirationUnit: TimeUnit): RecordPersister {
            return RecordPersister(fileSystem, expirationDuration, expirationUnit)
        }
    }
}
